### ISSUE TITLE
`vector_list_t` intended to only support `real_t` and `integer_t`?

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2662]
`igraph_vector_list_t` is set up only for reals and integers:

https://github.com/igraph/igraph/blob/bb2b55bd6704832e0d6596bb18ff90e2003290e3/include/igraph_vector_list.h#L38-L52

But the header still has macros referencing other types:

https://github.com/igraph/igraph/blob/bb2b55bd6704832e0d6596bb18ff90e2003290e3/include/igraph_vector_list.h#L58-L69

It is also lacking a copy constructor and an `update()` function.

What was your intention here @ntamas and how do we proceed?

Note that I was playing with completing the C++ wrapper, where this came up: https://github.com/igraph/igraph-cpp

**Comment by ntamas** [comment]
> What was your intention here @ntamas and how do we proceed?

I don't remember, but it is a non-goal to provide a full set of `vector_list_t` functions for all data types, so I'd just remove the extra macros from `igraph_vector_list.h` and be done with it. We've been bitten by this in the past with `igraph_array3_t` and so where we started providing data structures that we anticipated that they _may_ be used by others in the distant future, and it just ended up being a maintenance burden. Let's not add bloat to the code unless necessary. I guess there are far better options for generic vectors and vectors-of-vectors in the C and C++ ecosystem.
